### ISSUE TITLE
Consistent import style.

### ISSUE THREAD

**Issue by tomchristie** [encode/httpcore#966]
Prompted by https://github.com/encode/httpcore/pull/963#discussion_r1794432560.

I think we should probably switch to using a consistent import style throughout for all stdlib and third-party-package imports.

For example, switch [from this](https://github.com/encode/httpcore/blob/master/httpcore/_async/connection.py)...

```python
from __future__ import annotations

import itertools
import logging
import ssl
from types import TracebackType
from typing import Iterable, Iterator

from .._backends.auto import AutoBackend
from .._backends.base import SOCKET_OPTION, AsyncNetworkBackend, AsyncNetworkStream
from .._exceptions import ConnectError, ConnectTimeout
...
```

To this...

```python
from __future__ import annotations

import itertools
import logging
import ssl
import types
import typing

from .._backends.auto import AutoBackend
from .._backends.base import SOCKET_OPTION, AsyncNetworkBackend, AsyncNetworkStream
from .._exceptions import ConnectError, ConnectTimeout
...
```

**Comment by zrquan** [comment]
This might make some parts of the code a bit wordy.

For example [here](https://github.com/encode/httpcore/blob/127505be50c29d1f19bafe55f9b5a5b3df468baa/httpcore/_models.py#L20), become:
```python
HeadersAsSequence = typing.Sequence[typing.Tuple[typing.Union[bytes, str], typing.Union[bytes, str]]]
HeadersAsMapping = typing.Mapping[typing.Union[bytes, str], typing.Union[bytes, str]]
HeaderTypes = typing.Union[HeadersAsSequence, HeadersAsMapping, None]
```

**Comment by zrquan** [comment]
> ```python
> HeadersAsSequence = typing.Sequence[tuple[bytes | str, bytes | str]]
> HeadersAsMapping = typing.Mapping[bytes | str, bytes | str]
> HeaderTypes = typing.Union[HeadersAsSequence, HeadersAsMapping, None]
> ```

This code doesn't pass mypy's checks, and I'm still not sure why.

**Comment by tomchristie** [comment]
> This code doesn't pass mypy's checks, and I'm still not sure why.

mypy is fuzzy and non deterministic enough to let us know we've got a code smell.

Still, let's get those simple consistent imports in.

We can drop in a ByteOrStr piece of nonsense as an interim, while we work towards a tighter API there.
